lib/commands.js

```diff
@@ -358,13 +359,26 @@ var operations = module.exports.operations = {
     }
 
     var diff = compare(context.oldParameters, context.newParameters);
+
     if (!diff) {
       context.overrides.skipConfirmParameters = true;
       return context.next();
     }
 
+    if (context.overrides.diffs && context.overrides.diffs.parameters) {
+      var preapproved = context.overrides.diffs.parameters.filter(function(previous) {
+        return previous === diff;
+      }).length;
+
+      if (preapproved) {
```

Comment:
what if only one item in `context.overrides.diffs.parameters` matches the diff - is this a sufficient condition? if yes why?

Reply:
Yes, this is sufficient. Consider a deploy you're making to 10 regions, and the starting point is that 5 of those regions are in one state (A), 5 are in another state (B) (where "state" means some different set of parameter values). Your deploy is going to set ALL the stacks to have the same parameter values (C).

The first region you hit, you approve the parameter diff to change state A --> state C. The second region is a state B stack, so now you approve the parameter diff for state B --> state C. Now the array of diffs contains both the A->C and B->C strings. Every subsequent region falls into one of those 2 categories, so you won't get prompted any more.

In the history of regions you've hit, you've accumulated a set of diffs that you've approved. Any subsequent diffs only need to match 1 from that set to be "preapproved".

Reply:
Note that each string in the `context.overrides.diffs.parameters` array represents ALL the parameter changes from a single region -- not one string per parameter that changed.